tests/Microsoft.Bot.Builder.Core.Extensions.Tests/ShowTyping_MiddlewareTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Bot.Builder.Core.Extensions.Tests
+{
+    [TestClass]
+    public class ShowTyping_MiddlewareTests
+    {
+        [TestMethod]
+        [TestCategory("Middleware")]
+        public async Task ShowTyping_TestMiddleware_10SecondResponse_2SecondInterval()
+        {
+            TestAdapter adapter = new TestAdapter()
+                .Use(new ShowTypingMiddleware(500, 2000));
+            
+            await new TestFlow(adapter, async (context) =>
+                {
+                    Thread.Sleep(10000);
```

Comment:
This will sometimes fail, If I understand it correctly. 

Send an indicator every 2000 ms on one thread. 
Have another thread sleep for 10s. 

Every now and then, one will complete before the other. 

Also, consider making this quicker, just so the tests run faster. 10s is a long time for a test! :)

Reply:
I have updated the timings here to avoid the situation you described.